src/main/java/org/opentripplanner/routing/algorithm/filterchain/filter/GroupByFilter.java

```diff
@@ -1,48 +1,34 @@
-package org.opentripplanner.routing.algorithm.filterchain.filters;
+package org.opentripplanner.routing.algorithm.filterchain.filter;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.function.Function;
 import org.opentripplanner.model.plan.Itinerary;
-import org.opentripplanner.routing.algorithm.filterchain.ItineraryFilter;
+import org.opentripplanner.routing.algorithm.filterchain.ItineraryListFilter;
 import org.opentripplanner.routing.algorithm.filterchain.groupids.GroupId;
 
 
 /**
- * This filter group the itineraries using a group-id and filter each group
- * by the given {@code filter}. It ensure that {@code maxNumberOfItinerariesPrGroup} requirement
- * is meat by reducing each group down to the given limit.
+ * This filter groups the itineraries using a group-id and filter each group
+ * by the given {@code filter}.
  *
  * @see GroupId on how to group itineraries
  */
-public class GroupByFilter<T extends GroupId<T>> implements ItineraryFilter {
+public class GroupByFilter<T extends GroupId<T>> implements ItineraryListFilter {
 
-    private final String name;
-    private final int maxNumberOfItinerariesPrGroup;
     private final Function<Itinerary, T> groupingBy;
-    private final ItineraryFilter arrangeBy;
+    private final List<ItineraryListFilter> nestedFilters;
 
     public GroupByFilter(
-            String name,
             Function<Itinerary, T> groupingBy,
-            ItineraryFilter arrangeBy,
-            int maxNumberOfItinerariesPrGroup
+            List<ItineraryListFilter> nestedFilters
     ) {
-        this.name = name;
         this.groupingBy = groupingBy;
-        this.arrangeBy = arrangeBy;
-        this.maxNumberOfItinerariesPrGroup = maxNumberOfItinerariesPrGroup;
-    }
-
-    @Override
-    public final String name() {
-        return name;
+        this.nestedFilters = nestedFilters;
     }
 
     @Override
     public final List<Itinerary> filter(List<Itinerary> itineraries) {
-        if(itineraries.size() <= maxNumberOfItinerariesPrGroup) { return itineraries; }
-
```

Comment:
This prevented the groupByFilters to remove to many itineraries, I know we eventually want to do that, but if this is a refactoring - keeping things working as before, then where is the code making sure we keep all itineraries if the threshold is met?